django_gamification/models.py

```diff
@@ -109,6 +109,27 @@ def save(self, *args, **kwargs):
                 badge.points = self.points
                 badge.save()
 
+    def award(self, interface):
+        """
+        Awards this badge to a user.
+        """
+        badges = interface.badge_set.filter(badge_definition=self)
+        badge = None
+        # There should only be one. Otherwise something horrible has happened.
+        # We can handle not having one, where we just need to create one.
+        if len(badges) == 1:
+            badge = badges[0]
+            # Ignore previously-acquired badges.
+            if badge.acquired:
+                return
+        elif len(badges) == 0:
+            badge = Badge.objects.create_badge(self, interface)
+        else:
+            raise "Expected one badge per definition but found two."
```

Comment:
We can eliminate the need for this manual check by adding a `unique_together` constraint on `Badge` between the `badge_definition` and `interface` fields.